### DISCUSSION TITLE
Automerge only lockFileMaintenance PRs

### DISCUSSION THREAD

**Discussion by jmac105** [renovatebot/renovate#9528] [Request Help] [answered]
Hi,

We currently have automerge disabled at the root config level, and then inside packageRules have enabled it for devDependencies. I'd like to also enable it for lockFileMaintenance PRs but not totally sure which way to do this as I can think of 2 different approaches?

```
  ...
  "automerge": false,
  "lockFileMaintenance": {
    "enabled": true,
    "schedule": ["after 7pm on sunday"],
    "automerge": true,
    "requiredStatusChecks": null
  },
  "autodiscover": true, // Find repos rather than us explicitly listing them
  ...
```
 or
```
  ...
  "automerge": false,
  "lockFileMaintenance": {
    "enabled": true,
    "schedule": ["after 7pm on sunday"]
  },
  "autodiscover": true, // Find repos rather than us explicitly listing them
  "packageRules": [
    {
      "requiredStatusChecks": null,
      "automerge": true,
      "matchUpdateTypes": [
        "lockFileMaintenance"
      ]
    }
  ],
  ...
```

Do both of these configurations do the same thing? I wasn't sure if one is recommended over the other

Thanks!

**Comment by viceice** [comment, marked as answer]
Yes, there are multiple ways to solve same problem.

Automerge is disabled by default, so you don't need to explicit disable it. Only enable it where you like it to have. 

Best way is to always use package rule.

**Reply by jmac105** [reply]
Thanks @viceice . I've explicitly disabled it just for clarity to colleagues who are less familiar with the tool, I'll go with the 2nd option of using a packageRule
